### PR TITLE
Add onHeaders callback

### PR DESCRIPTION
**Pull request by yaslama**

### CODE REVIEW

File: lib/net/xhr.js [hunk 1]
```diff
@@ -1,5 +1,30 @@
 'use strict';
 
+const trim = ''.trim
```
Comment (senthilp) [inline lib/net/xhr.js]: @yaslama `jsonpipe` was written in ES5 and hence doesn't have a babel gulp task to transpile ES6 to ES5.  Can you please change `const` to `var` and any other ES6 reference.

**Comment by senthilp** [comment]
@yaslama Can you please complete the following to merge the pull request

1. Add unit tests for this in the [test file](https://github.com/eBay/jsonpipe/blob/master/test/jsonpipe.js)
1.  Check-in the built file [jsonpipe.js](https://github.com/eBay/jsonpipe/blob/master/jsonpipe.js)
1. Update the documentation in the [usage](https://github.com/eBay/jsonpipe#usage) and [options](https://github.com/eBay/jsonpipe#options) section

**Comment by yaslama** [comment]
Hi @senthilp, I updated the PR as you requested.

**Comment by senthilp** [comment]
@yaslama Thanks for the PR.

File: jsonpipe.js [hunk 2]
```diff
@@ -99,6 +99,31 @@ module.exports = {
...
+  var line;
+  var field;
+  var val;
+
+  lines.pop(); // trailing CRLF
+
+  for (var i = 0, len = lines.length; i < len; ++i) {
+    line = lines[i];
+    index = line.indexOf(':');
+    field = line.slice(0, index).toLowerCase();
```
Comment (senthilp) [inline jsonpipe.js]: @yaslama is the `toLowerCase()` needed? Can we just have it as the way it comes from the response?

**Comment by yaslama** [comment]
We can. But see https://groups.google.com/forum/#!topic/nodejs/1aug0Jq8rYY

**Comment by senthilp** [comment]
The thread makes sense. Lets leave it as such now.

**Pull request merged**